eng/pipelines/templates/steps/build-and-test.yml

```diff
@@ -74,23 +80,53 @@ steps:
       AZURE_VERSION_OVERRIDE_TESTS: ${{ parameters.TestVersionSupport }}
     condition: and(succeeded(), eq(variables['TestVersionSupport'], 'true'))
 
-  # we want to run this when TestFromSource isn't true
-  - task: Maven@3
-    ${{ if ne(parameters.VersionOverride, '') }}:
-      displayName: '${{ parameters.VersionOverride }} - Run tests'
-    ${{ else }}:
-      displayName: 'Run tests'
-    inputs:
-      mavenPomFile: pom.xml
-      options: '$(DefaultTestOptions) -DAZURE_TEST_DEBUG=$(IsDebug) -Dparallel-test-playback=${{ parameters.ParallelTestPlayback }} ${{ parameters.TestOptions }} -T ${{ parameters.TestParallelization }}'
-      mavenOptions: '$(MemoryOptions) $(LoggingOptions)'
-      javaHomeOption: 'JDKVersion'
-      jdkVersionOption: $(JavaTestVersion)
-      jdkArchitectureOption: 'x64'
-      publishJUnitResults: false
-      goals: ${{ parameters.TestGoals }}
-    env: ${{ parameters.TestEnvVars }}
-    condition: and(succeeded(), ne(variables['TestFromSource'], 'true'))
+  - ${{ if eq('true', parameters.UseFederatedAuth) }}:
+    # Remove preceeding "1." in the version specified by $(JavaTestVersion)
+    # e.g. 1.8 -> 8
+    # This is required to set JAVA_HOME correctly for the Java version
+    - pwsh: |
+        $javaTestVersion = "$(JavaTestVersion)" -replace '^\d*?\.', ''
```

Comment:
this is fine but `"$(JavaTestVesrion)".Substring(2)` might be clearer? It's always 1. right?